xonsh/parsers/base.py

```diff
@@ -126,9 +126,10 @@ def xonsh_superhelp(x, lineno=None, col=None):
     return xonsh_call('__xonsh_superhelp__', [x], lineno=lineno, col=col)
 
 
-def xonsh_regexpath(x, lineno=None, col=None):
+def xonsh_regexpath(x, pymode=False, lineno=None, col=None):
     """Creates the AST node for calling the __xonsh_regexpath__() function."""
-    return xonsh_call('__xonsh_regexpath__', [x], lineno=lineno, col=col)
+    return xonsh_call('__xonsh_regexpath__', args=[x, pymode], lineno=lineno,
```

Comment:
The problem here is that pymode, needs to be an AST node, not a normal Python value. I think if you add the following line above this one, everything will workout:

```python
pymode = ast.NameConstant(value=pymode, lineno=lineno, col_offset=col)
```